Sources/TensorFlow/Operators/Math.swift

```diff
@@ -1229,7 +1229,7 @@ public func softmax<T: TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
 @differentiable
 public func softmax<T: TensorFlowFloatingPoint>(_ x: Tensor<T>, alongAxis axis: Int) -> Tensor<T> {
   let xExp = exp(x)
-  return xExp / xExp.sum(alongAxes: Tensor<Int32>(Int32(axis)))
+  return xExp / xExp.sum(alongAxes: Tensor<Int32>(Int32(axis), on: xExp.device))
```

Comment:
I assume you're confident you caught all places where this is needed thanks to the fact that there are no implicit device transfers.

Reply:
Only mostly so.